src/axom/core/StaticArray.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_STATICARRAY_HPP_
+#define AXOM_STATICARRAY_HPP_
+
+#include "axom/config.hpp"           // for compile-time defines
+#include "axom/core/Macros.hpp"      // for axom macros
+#include "axom/core/StackArray.hpp"  // for StackArray
+
+namespace axom
+{
+/*!
+ * \accelerated
+ * \class StaticArray
+ *
+ * \brief Provides a wrapper for a StackArray with an additional size member.
+ *
+ * \tparam T the type of the values to hold.
```

Comment:
Is it important to note that the data type 'T' must have a default constructor that can run on a device?